providers/azureblob/src/main/java/org/jclouds/azure/storage/filters/SharedKeyLiteAuthentication.java

```diff
@@ -110,7 +175,27 @@ HttpRequest replaceDateHeader(HttpRequest request) {
       request = request.toBuilder().replaceHeaders(Multimaps.forMap(builder.build())).build();
       return request;
    }
-
+   
+   /**
+   * this is the method to parse container name and blob name from the HttpRequest. 
+   * applicable for the cases with SAS Authentication 
+   * 
+   */ 
+   public String[] cutUri (URI uri) throws NullPointerException, IllegalArgumentException {
+      String path = uri.getPath();
+      String[] result = path.split("/");
+      if (result.length < 2) {
+         throw new IllegalArgumentException();
```

Comment:
Add a message to the exception describing what's wrong.

Reply:
Done!